widgetssdk/src/test/java/com/glia/widgets/engagement/EngagementRepositoryTest.kt

```diff
@@ -477,6 +488,31 @@ class EngagementRepositoryTest {
         confirmVerified(incomingEngagementRequest)
     }
 
+    @Test
+    fun `engagementOutcome will emit value when outcome is requested`() {
+        val incomingEngagementRequest: IncomingEngagementRequest = mockk()
+        val outcomeSlot = slot<Consumer<EngagementRequest.Outcome>>()
+        every { incomingEngagementRequest.on(EngagementRequest.Events.OUTCOME, capture(outcomeSlot)) } returns Unit
+        val engagementOutcomeTest = repository.engagementOutcome.test()
+
+        engagementRequestCallbackSlot.captured.accept(incomingEngagementRequest)
+        outcomeSlot.captured.accept(EngagementRequest.Outcome.ACCEPTED)
+
+        engagementOutcomeTest.assertNoErrors().assertValue(EngagementRequest.Outcome.ACCEPTED)
+    }
+
+    @Test
+    fun `unsubscribe from engagement outcome event will happen when outcome is requested`() {
```

Comment:
"when outcome is requested" or "accepted"?

Reply:
In my understanding the outcome is received by SDK but it's not explicitly requested. Am I wrong?

Reply:
We can get it from the `visitor_state`. To do it, we need to subscribe for the `outcome`. But after we get it, we don't need the subscription anymore.